larray/core/array.py

```diff
@@ -7494,16 +7432,45 @@ def ndtest(shape, start=0, label_start=0, title='', dtype=int):
     a\\b  b1  b2  b3
      a1   0   1   2
      a2   3   4   5
+    >>> ndtest((2, 3), start=2)
+    a\\b  b0  b1  b2
+     a0   2   3   4
+     a1   5   6   7
+    >>> ndtest((2, 3), dtype=float)
+    a\\b   b0   b1   b2
+     a0  0.0  1.0  2.0
+     a1  3.0  4.0  5.0
+
+    Create test array by passing axes
+
+    >>> ndtest("nat=BE,FO;sex=M,F")
+    nat\\sex  M  F
+         BE  0  1
+         FO  2  3
+    >>> nat = Axis("nat=BE,FO")
+    >>> sex = Axis("sex=M,F")
+    >>> ndtest([nat, sex])
+    nat\\sex  M  F
+         BE  0  1
+         FO  2  3
     """
-    a = ndrange(shape, start=start, dtype=dtype, title=title)
+    # XXX: try to come up with a syntax where start is before "end".
+    # For ndim > 1, I cannot think of anything nice.
+    if isinstance(shape_or_axes, int):
+        shape_or_axes = (shape_or_axes,)
+    if isinstance(shape_or_axes, (list, tuple)) and all([isinstance(i, int) for i in shape_or_axes]):
+        axes_names = [chr(ord('a') + i) for i in range(len(shape_or_axes))]
+        label_ranges = [range(label_start, label_start + length) for length in shape_or_axes]
+        shape_or_axes = [Axis(['{}{}'.format(name, i) for i in label_range], name)
+                         for name, label_range in zip(axes_names, label_ranges)]
+    if isinstance(shape_or_axes, AxisCollection):
+        axes = shape_or_axes
+    else:
+        axes = AxisCollection(shape_or_axes)
     # TODO: move this to a class method on AxisCollection
-    assert a.ndim <= 26
-    axes_names = [chr(ord('a') + i) for i in range(a.ndim)]
-    label_ranges = [range(label_start, label_start + length)
-                    for length in a.shape]
-    new_axes = [Axis([name + str(i) for i in label_range], name)
-                for name, label_range in zip(axes_names, label_ranges)]
-    return a.set_axes(new_axes)
+    assert len(axes) <= 26
```

Comment:
Move this assert before `axes_names =`. If passing explicit axes, we should not prevent having as many axes as wanted.